core/src/data.rs

```diff
@@ -28,6 +28,8 @@ pub enum DataType {
     Int(i32),
     /// A 64-bit numeric value.
     BigInt(i64),
+    /// A 64-bit numeric ID, prefixed by its shard index (or 0 if unsharded).
+    ID(u32, i64),
```

Comment:
Can we check that this doesn't increase the size of `DataType`? I don't *think* it should, but worthwhile to check given how ubiquitous `DataType` is. From memory, the command to run is
```console
$ cargo rustc -Zprint-type-sizes
```
It generates a bunch of output, so pipe it to a file.

Reply:
Also, why is the sequence number an `i64` instead of a `u64`?

Reply:
Changed to `u64`. [This test](https://github.com/mit-pdos/distributary/blob/master/basics/src/data.rs#L1031) should make sure the `DataType` size hasn't changed.